brainframe_qt/api_utils/__init__.py

```diff
@@ -1,5 +1,22 @@
+import typing
+
+from PyQt5.QtCore import QObject
 from brainframe.api import BrainFrameAPI
-from .streaming import StreamManager, StreamManagerAPI
 
-# API instance that is later monkeypatched to be a singleton
-api: BrainFrameAPI = StreamManagerAPI()
+# Singleton API instance
+api = BrainFrameAPI()
+
+# Must come after api import
+from .streaming import StreamManager
+
+# Set using init_stream_manager before use
+_stream_manager: StreamManager = typing.cast(StreamManager, None)
+
+
+def init_stream_manager(*, parent: QObject) -> None:
+    global _stream_manager
+    _stream_manager = StreamManager(parent=parent)
+
+
+def get_stream_manager() -> StreamManager:
+    return _stream_manager
```

Comment:
Creating singleton QObjects is a bit of a hassle because they need a parent object.

Flutter has a really cool system called `InheritedWidget`/`Provider` that makes passing objects  like this around really easily. I'd like to implement something similar, but for now this is the only singleton object we have so I think this should be begrudgingly "ok".

If you have any better ideas, I'm all ears.

Reply:
I'll add this as a comment to the code if there aren't any better ideas

Reply:
I guess my only question would be- is there any case where you would want to get the StreamManager but not create it if it doesn't exist? 

I figure that a "get or create" type method might fit the needs better here.